File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hashmap=defaultdict(int)
        n=len(nums)
        for i in range(n):
            hashmap[nums[i]]+=1
        for key,value in hashmap.items():
            if value>n//2:
                print(key,value)
                return key

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int lim=(int) nums.length/2;
        HashMap<Integer,Integer> map =new HashMap<>();
        for(int num:nums){
            map.put(num,map.getOrDefault(num,0)+1);
        }
        for(Integer key:map.keySet()){
            if(map.get(key)>lim){
                return key;
            }
        }
        return -1;
    }
}